python/src/cloudstorage/cloudstorage_api.py

```diff
@@ -273,6 +275,185 @@ def listbucket(path_prefix, marker=None, prefix=None, max_keys=None,
+    Merges between 2 and 1024 files into one file. If any of the previous files had the merge
+    run on them it counts towards the 1024 limit.
+    Automatically breaks down the files into batches of 32.
+    This method is slower and could result in orphan files if it fails
+  Args:
+    list_of_files: list of file name strings or dictionaries with the following format:
+      {'file_name': REQUIRED name of the file to be merged. Do not include the bucket name,
+       'Generation': OPTIONAL Used to specify what version of a file to use,
+       'IfGenerationMatch': OPTIONAL Used to fail requests if versions don't match}
+    destination_file: Path to the desired output file. Must have the bucket in the path.
+    preserve_order: If true will not sort the files into natural order.
+    content_type: Used to specify the content-header of the output.
+                  If None will try to guess based off the first file.
+    retry_params: An api_utils.RetryParams for this call to GCS. If None,
+    the default one is used.
+    _account_id: Internal-use only.
+
+  Raises:
+    TypeError: If the dictionary for the file list is malformed
+    ValueError: If the number of files is outside the range of 2-1024
+    errors.NotFoundError: If any element in the file list is missing the 'file_name' key
+  '''
+
+  temp_file_suffix = '____MergeTempFile'
+
+  file_list, bucket, content_type = _validate_compose_list(
+                                      destination_file, list_of_files,
+                                   content_type, number_of_files=1024)
+
+  '''
+  Compose can only handle 32 files at a time. Breaks down the list into batches of 32
+  (this will only need to happen once, since the file_list size restriction is 1024 = 32 * 32)
+  '''
+  temp_list = []  # temporary storage for the filenames that store the merged segments of 32
+  if len(file_list) > 32:
+    temp_file_counter = 0
+    segments_list = [file_list[i:i + 32] for i in range(0, len(file_list), 32)]
+    file_list = []
+    for segment in segments_list:
+      if len(segment) > 1:
+        temp_file_name = destination_file + temp_file_suffix + str(temp_file_counter)
+        compose(segment, temp_file_name, content_type, retry_params, _account_id)
+        file_list.append({'file_name': temp_file_name.replace(bucket, '', 1)})
+        temp_file_counter += 1
+        temp_list.append(temp_file_name)
+      else:
+        file_list.append(segment[0])
+  # There will always be 32 or less files to merge at this point
+  compose(file_list, destination_file, content_type, retry_params, _account_id)
+  # grab all temp files that were created during the merging of segments of 32
+  temp_list = listbucket(destination_file + temp_file_suffix)
+  # delete all the now-unneeded temporary merge-files for the segments of 32 (if any)
+  for item in temp_list:
+    try:
+      delete(item.filename)
+    except errors.NotFoundError:
+      pass
+
+def file_exists(destination):
```

Comment:
can this be made "private" as well (_file_exists)?
Also, I think it should be enough just to open the file for read, no?